graylog2-web-interface/src/components/configurationforms/ConfigurationForm.tsx

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useState, useEffect, forwardRef, useImperativeHandle } from 'react';
+
+import BootstrapModalForm from 'components/bootstrap/BootstrapModalForm';
+import { ConfigurationFormField, TitleField } from 'components/configurationforms';
+
+import type { ConfigurationField } from './types';
+
+type Props = {
+  cancelAction: () => void,
+  configFields?: {
+    [key: string]: ConfigurationField,
+  },
+  children: React.ReactNode | null,
+  helpBlock: React.ReactNode | null,
+  includeTitleField: boolean,
+  submitAction: (data: any) => void,
```

Comment:
I think in this case we can define a generic type which extends an object and which can be defined when using the component. This way we could type `data` and  the `values` prop and  avoid using `any`.

`type Props<T extends object>`
...
`<ConfigurationForm<{...}> `